### PR TITLE
ci: Add workflow to build m1 conda binaries

### PR DESCRIPTION
**Pull request by seemethere**
Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

### CODE REVIEW

**Review by osalpekar — APPROVED** [review]

File: .github/workflows/build_conda_m1.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+          python-version: ${{ env.PYTHON_VERSION }}
+      - name: Setup base environment variables
+        run: |
+          echo "SOURCE_ROOT_DIR=${GITHUB_WORKSPACE}/${REPOSITORY}" >> "${GITHUB_ENV}"
+      - name: Build the wheel (bdist_wheel)
+        working-directory: ${{ inputs.repository }}
+        run: |
+          source "${BUILD_ENV_FILE}"
+          ${CONDA_RUN} conda build \
+            -c defaults \
```
Comment (osalpekar) [inline .github/workflows/build_conda_m1.yml:70]: Do we need the CUDATOOLKIT_CHANNEL as well?

File: .github/workflows/build_conda_m1.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+          path: test-infra
+      - uses: ./test-infra/.github/actions/setup-binary-builds
+        with:
+          repository: ${{ inputs.repository }}
+          ref: ${{ inputs.ref }}
+          setup-miniconda: false
+          python-version: ${{ env.PYTHON_VERSION }}
+      - name: Setup base environment variables
+        run: |
+          echo "SOURCE_ROOT_DIR=${GITHUB_WORKSPACE}/${REPOSITORY}" >> "${GITHUB_ENV}"
```
Comment (osalpekar) [inline .github/workflows/build_conda_m1.yml:64]: If there are pre/post steps to run prior to running the conda build (like FFMPEG build), where will those go?

File: .github/workflows/build_conda_m1.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      - name: Build the wheel (bdist_wheel)
```
Comment (osalpekar) [inline .github/workflows/build_conda_m1.yml:65]: nit: conda instead of wheel

**Review by atalman — APPROVED** [review]
LGTM

**Pull request merged**